File: mesosphere-js/src/utils/embedding_callable.ts
import type { EmbeddingMatrix } from "../types";

/**
 * Base callable interface for embedding providers.
 */
export abstract class EmbeddingCallable {
  abstract embed(texts: string[]): Promise<EmbeddingMatrix>;

  async call(texts: string[]): Promise<EmbeddingMatrix> {
    return this.embed(texts);
  }
}

File: mesosphere/utils/embedding_callable.py
"""Callable helper for embedding providers."""

from typing import List


class EmbeddingCallable:
    """Protocol-like helper that makes embedding providers callable."""

    def __call__(self, texts: List[str]) -> List[List[float]]:
        """Make embedding providers callable for compatibility with integrations."""

        return self.embed(texts)
